src/components/Modals/NewCourse/CourseConflictModal.vue

```diff
@@ -187,32 +184,32 @@ export default defineComponent({
     resolveConflicts() {
       // edit the requirements assigned to the course when editor saved on reqs with changes made
       for (let i = 0; i < this.selectedReqsPerConflict.length; i += 1) {
-        const conflict = this.selectedReqsPerConflict[i];
-        const originalConflict = this.originalReqsPerConflict[i];
-
-        conflict.forEach((selected, reqName) => {
-          if (this.isReqSelectable(reqName) && selected !== originalConflict.get(reqName)) {
-            toggleRequirementChoice(
-              this.courseUniqueId,
-              store.state.userRequirementsMap[reqName].id,
-              'acknowledgedCheckerWarningOptIn'
-            );
-          } else if (selected !== originalConflict.get(reqName)) {
-            toggleRequirementChoice(
-              this.courseUniqueId,
-              store.state.userRequirementsMap[reqName].id,
-              'optOut'
-            );
-          }
-        });
+        const userChoices = Array.from(this.selectedReqsPerConflict[i].entries());
+
+        const naturallyFulfilledRequirementIds = userChoices
+          .filter(([reqName, selected]) => selected && !this.isReqSelectable(reqName))
+          .map(([reqName]) => store.state.userRequirementsMap[reqName].id);
+        const arbitraryOptInRequirementIds = userChoices
```

Comment:
The "arbitrary" terminology is very confusing IMO but it's already become standard in many places. I think in the future it might be worthwhile to change it.